src/main/java/com/cloudbees/fm/jenkins/FeatureManagementConfigurationBuilder.java

```diff
@@ -217,7 +219,11 @@ private String getApiToken(String credentialsId) {
             return creds.stream().findFirst().map(c -> c.getSecret().getPlainText()).orElseThrow(() -> new RuntimeException("Could not find credential ID " + credentialsId));
         }
 
-        public ListBoxModel doFillApplicationIdAndNameItems(@QueryParameter String credentialsId) throws IOException {
+        @POST
+        public ListBoxModel doFillApplicationIdAndNameItems(@QueryParameter String credentialsId, @AncestorInPath Item item) throws IOException {
+            if (item != null) {
+                item.checkPermission(Permission.CONFIGURE);
+            }
             if (StringUtils.isBlank(credentialsId) || invalidCredentialIds.contains(credentialsId)) {
                 return null;
```

Comment:
Not changed in this PR but you should make sure the result is reasonable in this case. I don't know how the UI renders `null`.

Reply:
it renders it as an empty box, which works for me